backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -623,6 +457,489 @@ func initDBDriver(driverName string, initConnectionTimeout time.Duration) string
 	return sqlConfig
 }
 
+func isLegacySchema(db *gorm.DB) (bool, error) {
+	if !db.Migrator().HasTable(&model.Pipeline{}) {
+		glog.Infof("Pipelines table not found. Assuming fresh install.")
+		return false, nil
+	}
+	length, ok, err := getColumnLength(db, &model.Pipeline{}, "UUID")
+	if err != nil {
+		return false, fmt.Errorf("detect schema version: %w", err)
+	}
+	return !ok || length > 64, nil
+}
+
+func runLegacyUpgradeFlow(db *gorm.DB, dialect SQLDialect) error {
+	glog.Infof("Detected legacy schema. Running upgrade flow.")
+	// Step 1: decide whether to backfill pipeline_versions
+	// If pipeline_versions table is introduced into DB for the first time,
+	// it needs initialization or data backfill.
+	var tableNames []string
+	initializePipelineVersions := true
+	db.Raw(`show tables`).Pluck("Tables_in_mlpipeline", &tableNames)
+	for _, tableName := range tableNames {
+		if tableName == "pipeline_versions" {
+			initializePipelineVersions = false
+			break
+		}
+	}
+	// Step 2: block upgrade if legacy data too long
+	if err := runPreflightLengthChecks(db, dialect, validation.LengthSpecs); err != nil {
+		return fmt.Errorf("preflight length check failed: %w", err)
+	}
+
+	// Step 3: drop all foreign key constraints which can block shrinking columns
+	if err := dropAllFKConstraints(db, dialect.Name); err != nil {
+		return fmt.Errorf("drop foreign key constraints failed: %w", err)
+	}
```

Comment:
can we remove this and the associated function calls now that we are just dropping the specific constraints via dropLegacyIndexes?

Reply:
ah I misread the calls - you are correct. Thanks.